docs/default.nix

```diff
@@ -119,13 +121,83 @@ let
-            '';
+
+            readme =
+              let
+                path = "${inputs.self}/modules/${module}/README.md";
+
+                # This doesn't count as IFD because ${inputs.self} is a flake input
+                mainText =
+                  if builtins.pathExists path then
+                    builtins.readFile path
+                  else
+                    ''
+                      # ${module}
+                      > [!NOTE]
+                      > This module doesn't include any additional documentation.
+                      > You can browse the options it provides below.
+                    '';
+
+                inherit (metadata.${module}) maintainers;
+
+                # Render a maintainer's name and a link to the best contact
+                # information we have for them.
+                #
+                # The reasoning behind the order of preference is as follows:
+                #
+                # - GitHub:
+                #   - May link to multiple contact methods
+                #   - More likely to have up-to-date information than the
+                #     maintainers list
+                #   - Protects the email address from crawlers
+                # - Email:
+                #   - Very commonly used
+                # - Matrix:
+                #   - Only other contact method in the schema
+                #     (as of March 2025)
+                # - Name:
+                #   - If no other information is available, then just show
+                #     the maintainer's name without a link
+                renderMaintainer =
+                  maintainer:
+                  if maintainer ? github then
+                    "[${maintainer.name}](https://github.com/${maintainer.github})"
+                  else if maintainer ? email then
+                    "[${maintainer.name}](mailto:${maintainer.email})"
+                  else if maintainer ? matrix then
+                    "[${maintainer.name}](https://matrix.to/#/${maintainer.matrix})"
+                  else
+                    maintainer.name;
+
+                joinItems =
+                  items:
+                  if builtins.length items == 0 then
+                    ""
+                  else if builtins.length items == 1 then
+                    builtins.head items
+                  else if builtins.length items == 2 then
+                    "${builtins.head items} and ${builtins.elemAt items 1}"
+                  else
+                    "${builtins.head items}, ${joinItems (builtins.tail items)}";
+
+                renderedMaintainers = joinItems (map renderMaintainer maintainers);
+
+                maintainersText =
+                  if maintainers == [ ] then
+                    ''
+                      > [!WARNING]
+                      > This module has no assigned maintainer. It may not be
+                      > up-to-date for the latest version of the software being
+                      > themed.
```

Comment:
Open to suggestions on what to say / not say here.

Reply:
> This module has no assigned maintainer.

What about simply:

```
This module has no dedicated maintainers.
```

The "dedicated" word implies that the module is generally maintained but there is no actual user of the module that is also involved in its development.

> It may not be up-to-date for the latest version of the software being themed.

What about removing this, since it is somewhat implied by not having a dedicated maintainer?

Reply:
Yes, that makes sense, implying that it might be maintained by the community but not any specific person. Then we also don't say that a module is out of date when it may in fact be kept up to date in that way.

Reply:
> Should we include a link to the page explaining how to add a maintainer?

Good idea. The linked section also defines "module maintainer" in more depth:

```
This module has no [dedicated
maintainers](https://stylix.danth.me/modules.html#maintainers).
```